app/models/variant_override.rb

```diff
@@ -52,11 +52,7 @@ def move_stock!(quantity)
       return
     end
 
-    if quantity > 0
-      increment! :count_on_hand, quantity
-    elsif quantity < 0
-      decrement! :count_on_hand, -quantity
-    end
+    update!(count_on_hand: (count_on_hand || 0) + quantity)
```

Comment:
This is not the same. An `update` call is not relative to the database values. It also runs validations and can fail there. I know that's the reason why Rubocop warns about this method but I'm not sure if our code relies on this. Ideally we would have other locking in place to avoid race conditions. We do lock during checkout but not when an admin is changing orders. So this might be a bit risky.

As sad as it is, I would keep the old code for now. We want to replace variant overrides anyway. Maybe place a `disable:rubocop` on this method instead?

Reply:
I reverted to the previous version, with a rubocop comment and a link to this conversation.